clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java

```diff
@@ -1011,33 +1036,47 @@ public void completeIfEmpty() {
         }
     }
 
-    static class Pair<V> {
+    static class Tuple<V> {
         private V asyncRequest;
-        private V syncRequest;
+        private Queue<V> syncRequestQueue;
+        private V closeRequest;
 
-        public Pair(V asyncRequest, V syncRequest) {
+        public Tuple(V asyncRequest, Queue<V> syncRequestQueue, V closeRequest) {
             this.asyncRequest = asyncRequest;
-            this.syncRequest = syncRequest;
+            this.syncRequestQueue = syncRequestQueue;
+            this.closeRequest = closeRequest;
         }
 
         public void setAsyncRequest(V asyncRequest) {
             this.asyncRequest = asyncRequest;
         }
 
-        public void setSyncRequest(V second) {
-            this.syncRequest = second;
+        public void setSyncRequestQueue(Queue<V> syncRequestQueue) {
```

Comment:
You could encapsulate this queue entirely in this class. For example, `addSyncRequest` could create the queue if it's not yet been created.

Reply:
Yeah makes sense, I have updated the addSyncRequest() method. Still the getSyncRequestQueue() method exists though as we need the queue contents when we process the acknowledgements. Here will renaming the method work, or is it fine as it is?

Reply:
That's fine. It just seemed weird forcing the caller to make and set the queue.